### DISCUSSION TITLE
How to learn about Roslyn

### DISCUSSION THREAD

**Discussion by paxhujing** [dotnet/roslyn#74350] [Q&A] [answered]
Can recommend any books or blogs?

**Comment by 333fred** [comment, marked as answer]
Depends on what you want to get out of learning Roslyn. Are you looking to write analyzers? If so, https://learn.microsoft.com/en-us/visualstudio/extensibility/getting-started-with-roslyn-analyzers?view=vs-2022. For source generators, https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md and https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.cookbook.md. If there's something else, https://github.com/dotnet/roslyn/tree/main/docs/wiki has a bunch of design docs. We're also active in the roslyn channel of https://discord.gg/csharp, and are more than happy to talk about specific questions there.

**Reply by paxhujing** [reply]
Thanks ! I need to write a code generator and profiler.

**Comment by WeihanLi** [comment]
We may need some `deep dive into Roslyn` series
